server/src/main/java/io/deephaven/server/session/SessionService.java

```diff
@@ -163,11 +170,13 @@ public SessionState newSession(final AuthContext authContext) {
      * @return the most recent token expiration
      */
     public TokenExpiration refreshToken(final SessionState session) {
-        return refreshToken(session, false);
+        TokenExpiration tokenExpiration = checkTokenAndRotate(session, false);
+        sessionListener.onSessionRefresh(session);
```

Comment:
Do we still want to invoke "onSessionRefresh" when tokenExpiration is null (ie, already expired)?

Reply:
Also, what about cases where the token wasn't actually refreshed?